pytorch_forecasting/data/timeseries.py

```diff
@@ -869,16 +869,18 @@ def __getitem__(self, idx: int) -> Tuple[Dict[str, torch.Tensor], torch.Tensor]:
             sequence_length >= self.min_prediction_length
         ), "Sequence length should be at least minimum prediction length"
         # determine prediction/decode length and encode length
-        decoder_length = min(
-            time[-1] - (self.min_prediction_idx - 1),
-            self.max_prediction_length,
-            sequence_length - self.min_encoder_length,
-        )
+        decoder_length = int(min(
```

Comment:
I would rather not cast this here because the user could still pass floats and then they are unpredictably converted/rounded to integers. Better to check at initialization that passed values are of the expected format.